### PR TITLE
Adding a CI job to push kubemci image

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://github.com/kubernetes/test-infra/issues/6624

cc @G-Harmon @BenTheElder @krzyzacy

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -597,6 +597,16 @@
       "sig-cluster-lifecycle"
     ]
   },
+  "ci-kubemci-image-push": {
+    "args": [
+      "make",
```
Comment (nikhiljindal) [inline jobs/config.json]: I have no idea about how to configure this to run make using the Makefile in `images/kubemci/Makefile`?
Any pointers?

Comment (BenTheElder) [reply jobs/config.json]: `"make", "-C", "images/kubemci", "push"`

Comment (BenTheElder) [reply jobs/config.json]: less opaquely: `make` accepts `-C subdir` https://www.gnu.org/software/make/manual/make.html#Options-Summary

Comment (nikhiljindal) [reply jobs/config.json]: Thanks for the super quick reply!!
Done

**Comment by BenTheElder** [comment]
Looks like you need an CI ("periodic") job entry in `prow/config.yaml`

File: prow/config.yaml [hunk 2]
```diff
@@ -436,6 +436,36 @@ presubmits:
         hostPath:
           path: /mnt/disks/ssd0/docker-graph
   GoogleCloudPlatform/k8s-multicluster-ingress:
+  - name: ci-kubemci-image-push
```
Comment (krzyzacy) [inline prow/config.yaml]: you probably don't want it to be a presubmit job..

Comment (nikhiljindal) [reply prow/config.yaml]: Right!! Thanks for pointing it out!
Moved to postsubmit.

File: prow/config.yaml [hunk 3]
```diff
@@ -5088,6 +5088,38 @@ postsubmits:
...
           path: /mnt/disks/ssd0/docker-graph
 
+  GoogleCloudPlatform/k8s-multicluster-ingress:
+  - name: ci-kubemci-image-push
+    agent: kubernetes
+    branches:
+    - master
+    labels:
+      preset-service-account: true
+      preset-k8s-ssh: true
```
[the remark this answers is not in the document]
Comment (nikhiljindal) [reply prow/config.yaml]: removed :)

**Comment by nikhiljindal** [comment]
@BenTheElder @krzyzacy 
This doesnt seem to be working. 

I was expecting to see a new run every time a PR merges in https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress, but I dont see any run here: https://k8s-testgrid.appspot.com/sig-multicluster-kubemci#kubemci-image-push.

Any ideas on how can I go about debugging this? It seems to me that the job is not even being triggered (since there is no run on testgrid), which is surprising.

**Comment by krzyzacy** [comment]
http://prow.k8s.io/?repo=GoogleCloudPlatform%2Fk8s-multicluster-ingress&type=postsubmit

looks the latest run is busted and the pod is already gone.. can you kick off a new one and let me know?

File: prow/config.yaml [hunk 4]
```diff
@@ -5088,6 +5088,37 @@ postsubmits:
...
+    agent: kubernetes
+    branches:
+    - master
+    labels:
+      preset-service-account: true
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20180212-83c830734-master
+        args:
+        - "--root=/go/src/"
```
Comment (krzyzacy) [inline prow/config.yaml]: ahhhh... whoops :-) add `--repo=github.com/GoogleCloudPlatform/k8s-multicluster-ingress` :-)

Comment (nikhiljindal) [reply prow/config.yaml]: https://github.com/kubernetes/test-infra/pull/6825

**Comment by nikhiljindal** [comment]
The job is atleast running now.
It builds the image fine, but fails at the push step: http://prow.k8s.io/log?job=ci-kubemci-image-push&id=6 with a vague EOF error. Wondering if its a permission issue.

@krzyzacy @BenTheElder Have you seen this error before? Any idea?

**Comment by krzyzacy** [comment]
eh, I cannot get onto deck now, can you paste a full log here?

**Comment by nikhiljindal** [comment]
Copied the logs here: https://gist.github.com/nikhiljindal/fa6e3b3a7521f0366eed9c192361dcb0

**Comment by nikhiljindal** [comment]
We finally got a successful run with https://github.com/kubernetes/test-infra/pull/6839: https://k8s-testgrid.appspot.com/sig-multicluster-kubemci#kubemci-image-push

Thanks @krzyzacy & @BenTheElder for all the help!

**Pull request merged**